src/main/java/vowelExtract/stringVowel.java

```diff
@@ -0,0 +1,25 @@
+package vowelExtract;
+import java.util.Scanner;  // Import statement for Scanner
+
+public class stringVowel {
+
+    public static String Vowel(String str) {
+        StringBuilder s = new StringBuilder();  // No need to initialize with ""
+
+        for (int i = 0; i < str.length(); i++) {
+            char a = str.charAt(i);
+            // Use String.indexOf() for compact vowel checking
+            if ("aeiouAEIOU".indexOf(a) != -1) {
+                s.append(a);
+            }
+        }
+        return s.toString();
+    }
+
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        String str = sc.nextLine();
```

Comment:
1. For console applications, use Scanner for input. 
2. Alternatively, consider using javax.swing.JOptionPane.showInputDialog() for GUI-based input.

Reply:
So I must use javax.swing.JOptionPane.showInputDialog() in this case right ??

Reply:
Also can you check the code for test as I have written it for the first time